nomad/rpc.go

```diff
@@ -528,6 +528,26 @@ func (r *rpcHandler) forwardRegion(region, method string, args interface{}, repl
 	return r.connPool.RPC(region, server.Addr, server.MajorVersion, method, args, reply)
 }
 
+func (r *rpcHandler) getServer(region, serverID string) (*serverParts, error) {
+	// Bail if we can't find any servers
+	r.peerLock.RLock()
```

Comment:
You're missing an RUnlock for the non-error path. You should be able to just put unlock in a defer and remove the explicit call:

```suggestion
	r.peerLock.RLock()
	defer r.peerLock.RUnlock()
```

Reply:
woops thanks